lib/data/external/AzureClient.js

```diff
@@ -11,11 +11,12 @@ const azureMpuUtils = s3middleware.azureHelper.mpuUtils;
 
 class AzureClient {
     constructor(config) {
-        this._azureBlobEndpoint = config.azureBlobEndpoint;
-        this._azureBlobSAS = config.azureBlobSAS;
+        this._azureStorageCredentials = config.azureStorageCredentials;
         this._azureContainerName = config.azureContainerName;
-        this._client = azure.createBlobServiceWithSas(
-          this._azureBlobEndpoint, this._azureBlobSAS);
+        this._client = azure.createBlobService(
+            this._azureStorageCredentials.storageAccountName,
+            this._azureStorageCredentials.storageAccessKey);
+        this._azureStorageEndpoint = this._client.host.primaryHost;
```

Comment:
are we relying on a documented property of the azure node sdk here?  do they have a method for the endpoint?  is this the best way to do this?

Reply:
Doesn't seem documented, they also don't have a method (they just access the StorageServiceClient.host.primaryHost property to get the value when sending a request). Maybe it's safest to require the blob endpoint explicitly again?

Reply:
do you have to tell the client the endpoint?  since it has this primary concept loaded, why do you need to tell it what it knows?

Reply:
we don't need it for the blob storage client, there is so far one case where we use it:
for the copy source uri https://github.com/scality/S3/pull/968/files#diff-96a8cd65c80ea7fb9b908a81388494d4R366
We always use a put object if copying from one account to another (instead of copy object) so the storage endpoint for the copy source uri should be the same as the destination endpoint